File: thirdMax.java
//22
class Solution {
    public int thirdMax(int[] nums) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
        for(int i = 0; i<nums.length; ++i){
            /*
            System.out.println(pq.peek());
            if(pq.size()>0)
                System.out.println("Size:" + pq.size());
            else
                System.out.println("Size:0");*/
            if(!pq.contains(nums[i])){
                if(pq.size()<3)
                    pq.add(nums[i]);
                else if(pq.size() == 3){
                    if((nums[i] > pq.peek())){
                        pq.poll();
                        pq.add(nums[i]);                    
                    }
                }
            }
        }
        if(pq.size() < 3){
            while(pq.size() != 1)
                pq.poll();
            
        }
        return pq.peek();
     }
}

File: thirdMax.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>

#define leng 4
using namespace std;


//99.33
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        unordered_map<int, int> map;
        priority_queue<int, vector<int>, greater<int>> pq;
        for(int i = 0; i<nums.size(); ++i){
            if(map[nums[i]]++ == 0){
                if(pq.size() < 3)
                    pq.push(nums[i]);
                else if(pq.size() == 3 && nums[i] > pq.top()){
                    pq.pop();
                    pq.push(nums[i]);
                }
            }
        }
        
        while(pq.size() < 3 && pq.size() != 1)
            pq.pop();
        
        return pq.top();
            
    }
};